lametro/events.py

```diff
@@ -70,8 +87,24 @@ def _find_partner(self, event):
 
             return None
 
+    def search(self, route, item_key, search_conditions):
+        search_url = self.BASE_URL + route
+
+        params = {"$filter": search_conditions}
+
+        # Add token to requests to the events endpoint, so we can capture SAP events.
+        if route == "/events/" and TOKEN:
+            params["token"] = TOKEN
```

Comment:
This is the only material change.

Reply:
we don't need this token for eventitems? or we will not be visiting evenitems?

is ther any reason not to follow the pattern in bills and use `self.params` so that the token is always provided?